cxx/irm.cc

```diff
@@ -8,21 +8,25 @@
 #include <functional>
 #include <variant>
 
+template <typename T>
+CleanRelation<T>* make_clean_relation(
+    Distribution<T>* unused_distribution,
+    const std::string& name,
+    const std::string& distribution_spec,
+    const std::vector<Domain*>& doms) {
+  return new CleanRelation<T>(name, distribution_spec, doms);
+}
+
 RelationVariant clean_relation_from_spec(const std::string& name,
-                                         const DistributionSpec& spec,
+                                         const std::string& distribution_spec,
                                          const std::vector<Domain*>& doms) {
-  switch (spec.observation_type) {
-    case ObservationEnum::bool_type:
-      return new CleanRelation<bool>(name, spec, doms);
-    case ObservationEnum::double_type:
-      return new CleanRelation<double>(name, spec, doms);
-    case ObservationEnum::int_type:
-      return new CleanRelation<int>(name, spec, doms);
-    case ObservationEnum::string_type:
-      return new CleanRelation<std::string>(name, spec, doms);
-    default:
-      assert(false && "Unsupported observation type.");
-  }
+  std::mt19937 prng;
+  DistributionVariant dv = get_distribution(distribution_spec, &prng);
```

Comment:
(Clarification: I support getting rid of `ObservationEnum`, I don't like that it's redundant with the distribution data type. As of now, I still think it's the least-bad solution we have, and we can continue to think about better ways to work around it.)

Reply:
Fixed the memory leak.

With either the current or proposed approach, we are dealing with lots of raw pointers to Distributions and Emissions, and the ownership of those is rarely annotated in the code.  So I think we will want to think about ways to avoid that in the future (maybe switch to Rust?  :) ), but I think it is almost entirely orthogonal to using ObservationEnums or not.

Reply:
More raw pointers creates more bug surface than fewer raw pointers, so I don't think it's orthogonal (I think often about switching to Rust :) )